### PR TITLE
ROX-12862: Enable postgres by default on local deploys only

### PR DESCRIPTION
**Pull request by theencee**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Review by dashrews78 — APPROVED** [review]
LGTM and I'm glad I looked as I'll need to explicitly turn it off for a test I'm working on.

**Review by dashrews78 — CHANGES_REQUESTED** [review]
actually I take that back.  You will need to turn it off for the `gke_upgrade_tests.py` too.

File: tests/e2e/run-ui-e2e.sh [hunk 1]
```diff
@@ -22,6 +22,11 @@ test_ui_e2e() {
     remove_existing_stackrox_resources
     setup_default_TLS_certs
 
+    [[ "$tag" =~ $RELEASE_RC_TAG_BASH_REGEX ]] || is_nightly_run || {
+        info "Disabling postgres as this is a release, RC or nightly build"
```
Comment (c-du) [inline tests/e2e/run-ui-e2e.sh:26]: Not familiar with this. but why limit Postgres testing  to dev builds?

Comment (theencee) [reply tests/e2e/run-ui-e2e.sh:26]: See this thread. Request from UI team: https://srox.slack.com/archives/C3W3U0RGW/p1664977840406369?thread_ts=1664927011.086369&cid=C3W3U0RGW

**Review by c-du — APPROVED** [review]

**Comment by theencee** [comment]
/test aro-qa-e2e-tests eks-qa-e2e-tests gke-kernel-qa-e2e-tests gke-nongroovy-e2e-tests gke-postgres-qa-e2e-tests gke-postgres-scale-tests gke-postgres-upgrade-tests gke-qa-e2e-tests gke-race-condition-qa-e2e-tests gke-scale-tests gke-ui-e2e-tests gke-upgrade-tests gke-version-compatibility-tests local-roxctl-tests openshift-newest-operator-e2e-tests openshift-newest-qa-e2e-tests openshift-oldest-operator-e2e-tests openshift-oldest-qa-e2e-tests openshift-penultimate-qa-e2e-tests osd-aws-qa-e2e-tests osd-gcp-qa-e2e-tests rosa-qa-e2e-tests

**Review by dashrews78 — APPROVED** [review]
LGTM

File: scripts/ci/jobs/gke_nongroovy_e2e_tests.py [hunk 2]
```diff
@@ -13,6 +13,9 @@
 # set required test parameters
 os.environ["ORCHESTRATOR_FLAVOR"] = "k8s"
 
+# don't use postgres
+os.environ["ROX_POSTGRES_DATASTORE"] = "false"
```
Comment (rhybrillou) [inline scripts/ci/jobs/gke_nongroovy_e2e_tests.py:17]: Should we plan to have postgres nongroovy tests?

File: scripts/ci/jobs/gke_race_condition_qa_e2e_tests.py [hunk 3]
```diff
@@ -15,4 +15,7 @@
 os.environ["CENTRAL_DB_IMAGE_TAG"] = os.environ["STACKROX_BUILD_TAG"]
 os.environ["USE_LOCAL_ROXCTL"] = "true"
 
+# don't use postgres
+os.environ["ROX_POSTGRES_DATASTORE"] = "false"
```
Comment (rhybrillou) [inline scripts/ci/jobs/gke_race_condition_qa_e2e_tests.py:19]: Should we plan to have postgres race condition tests ?

**Review by rhybrillou — APPROVED** [review]

File: scripts/ci/jobs/gke_nongroovy_e2e_tests.py (hunk 2 above)
Comment (theencee) [reply scripts/ci/jobs/gke_nongroovy_e2e_tests.py:17]: Yes :) (to both)

**Pull request merged**